packages/perseus/src/widgets/dropdown/serialize-dropdown.test.ts

```diff
@@ -0,0 +1,134 @@
+                        placeholder: "Choose",
+                        choices: [
+                            {
+                                content: "Correct",
+                                correct: true,
+                            },
+                            {
+                                content: "Incorrect",
+                                correct: false,
+                            },
+                        ],
+                    },
+                },
+            },
+        });
+        const item = generateTestPerseusItem({question});
+        return item;
+    }
+
+    beforeAll(() => {
+        registerAllWidgetsForTesting();
+    });
+
+    let userEvent: UserEvent;
+    beforeEach(() => {
+        userEvent = userEventLib.setup({
+            advanceTimers: jest.advanceTimersByTime,
+        });
+
+        jest.spyOn(Dependencies, "getDependencies").mockReturnValue(
+            testDependencies,
+        );
+    });
+
+    afterEach(() => {
+        // The Renderer uses a timer to wait for widgets to complete rendering.
+        // If we don't spin the timers here, then the timer fires in the test
+        // _after_ and breaks it because we do setState() in the callback,
+        // and by that point the component has been unmounted.
+        act(() => jest.runOnlyPendingTimers());
+    });
+
+    it("should serialize the current state", async () => {
+        // Arrange
+        const {renderer} = renderQuestion(generateBasicDropdown());
+
+        await userEvent.click(
+            screen.getByRole("combobox", {name: "Select an answer"}),
+        );
+        await userEvent.click(screen.getByRole("option", {name: "Correct"}));
+
+        // Act
+        const state = renderer.getSerializedState();
+
+        // Assert
+        expect(state).toEqual({
+            question: {
+                "dropdown 1": {
+                    choices: ["Correct", "Incorrect"],
+                    placeholder: "Choose",
+                    // selected is added to Dropdown props and
+                    // represents user input
+                    selected: 1,
+                },
+            },
+            hints: [],
+        });
+    });
+
+    it("should restore serialized state", () => {
+        // Arrange
+        const {renderer} = renderQuestion(generateBasicDropdown());
+
+        // Act
+        act(() =>
+            renderer.restoreSerializedState({
+                question: {
+                    "dropdown 1": {
+                        choices: ["Correct", "Incorrect"],
+                        placeholder: "Choose",
+                        selected: 2,
+                    },
+                },
+                hints: [],
+            }),
+        );
+
+        const userInput = renderer.getUserInput();
+
+        // Assert
+        // `value` would be 0 if we didn't properly restore serialized state
+        expect(userInput).toEqual({"dropdown 1": {value: 2}});
```

Comment:
🤔  Non-blocking, but I thought I'd mention this. Given that RTL is all about testing user-facing behaviour, would there be value in also asserting that the Dropdown has the restored option selected?